src/components/TxConfirmationView/NetworkMismatch/index.tsx

```diff
@@ -132,17 +85,26 @@ export default function NetworkMismatch({
                     params: [{ chainId: `0x${neededChainId.toString(16)}` }],
                   });
                 } catch (switchError) {
-                  if (config && switchError.code === 4902) {
+                  if (neededNetworkConfig && switchError.code === 4902) {
                     try {
                       await (window as any).ethereum?.request({
                         method: 'wallet_addEthereumChain',
                         params: [
                           {
                             chainId: `0x${neededChainId.toString(16)}`,
-                            chainName: config.name,
-                            nativeCurrency: config.nativeCurrency,
-                            rpcUrls: [...publicJsonRPCUrl, publicJsonRPCWSUrl],
-                            blockExplorerUrls: config.explorerUrls,
+                            chainName: neededNetworkConfig.name,
+                            nativeCurrency: {
+                              symbol: neededNetworkConfig.baseAsset.toUpperCase(),
+                              decimals: 18,
```

Comment:
for now i hardcoded the 18 here, might make sense to move it to network config instead though